src/OracleService/OracleService.cs

```diff
@@ -0,0 +1,489 @@
+                OnStop();
+        }
+
+        private async void OnTimer(object state)
+        {
+            List<ulong> outOfDate = new List<ulong>();
+            foreach (var (id, task) in pendingQueue)
+            {
+                var span = TimeProvider.Current.UtcNow - task.Timestamp;
+                if (span > TimeSpan.FromSeconds(RefreshInterval) && span < TimeSpan.FromSeconds(RefreshInterval * 2))
+                {
+                    List<Task> tasks = new List<Task>();
+                    foreach (var account in wallet.GetAccounts())
+                        if (task.BackupSigns.TryGetValue(account.GetKey().PublicKey, out byte[] sign))
+                            tasks.Add(SendResponseSignatureAsync(id, sign, account.GetKey()));
+                    await Task.WhenAll(tasks);
+                }
+                else if (span > Settings.Default.MaxTaskTimeout)
+                {
+                    outOfDate.Add(id);
+                }
+            }
+            foreach (ulong requestId in outOfDate)
+                pendingQueue.TryRemove(requestId, out _);
+            foreach (var (key, value) in finishedCache)
+                if (TimeProvider.Current.UtcNow - value > TimeSpan.FromDays(3))
+                    finishedCache.TryRemove(key, out _);
+
+            if (!cancelSource.IsCancellationRequested)
+                timer?.Change(RefreshInterval, Timeout.Infinite);
+        }
+
+        [RpcMethod]
+        public JObject SubmitOracleResponse(JArray _params)
+        {
+            if (stopped || !started) throw new InvalidOperationException();
+            ECPoint oraclePub = ECPoint.DecodePoint(Convert.FromBase64String(_params[0].AsString()), ECCurve.Secp256r1);
+            ulong requestId = (ulong)_params[1].AsNumber();
+            byte[] txSign = Convert.FromBase64String(_params[2].AsString());
+            byte[] msgSign = Convert.FromBase64String(_params[3].AsString());
+
+            if (finishedCache.ContainsKey(requestId)) throw new RpcException(-100, "Request has already finished");
+
+            using (SnapshotView snapshot = Blockchain.Singleton.GetSnapshot())
+            {
+                var oracles = NativeContract.RoleManagement.GetDesignatedByRole(snapshot, Role.Oracle, snapshot.Height + 1);
+                if (!oracles.Any(p => p.Equals(oraclePub))) throw new RpcException(-100, $"{oraclePub} isn't an oracle node");
+                if (NativeContract.Oracle.GetRequest(snapshot, requestId) is null)
+                    throw new RpcException(-100, "Request is not found");
+                var data = Neo.Helper.Concat(oraclePub.ToArray(), BitConverter.GetBytes(requestId), txSign);
+                if (!Crypto.VerifySignature(data, msgSign, oraclePub)) throw new RpcException(-100, "Invalid sign");
+
+                AddResponseTxSign(snapshot, requestId, oraclePub, txSign);
+            }
+            return new JObject();
```

Comment:
Why return an empty `JObject`?

Reply:
It needs a return value, otherwise it'll throw an exception.